src/sandmastery/Java/leaf/cosmere/sandmastery/common/manifestation/MasteryCushion.java

```diff
@@ -0,0 +1,54 @@
+package leaf.cosmere.sandmastery.common.manifestation;
+
+import leaf.cosmere.api.Manifestations;
+import leaf.cosmere.api.Taldain;
+import leaf.cosmere.api.math.VectorHelper;
+import leaf.cosmere.api.spiritweb.ISpiritweb;
+import leaf.cosmere.common.cap.entity.SpiritwebCapability;
+import leaf.cosmere.sandmastery.client.SandmasteryKeybindings;
+import leaf.cosmere.sandmastery.common.capabilities.SandmasterySpiritwebSubmodule;
+import leaf.cosmere.sandmastery.common.utils.MiscHelper;
+import net.minecraft.world.entity.LivingEntity;
+import net.minecraft.world.phys.Vec3;
+
+public class MasteryCushion extends SandmasteryManifestation{
+    public MasteryCushion(Taldain.Mastery mastery) {
+        super(mastery);
+    }
+
+    @Override
+    public void tick(ISpiritweb data)
+    {
+        SpiritwebCapability playerSpiritweb = (SpiritwebCapability) data;
+        SandmasterySpiritwebSubmodule submodule = (SandmasterySpiritwebSubmodule) playerSpiritweb.spiritwebSubmodules.get(Manifestations.ManifestationTypes.SANDMASTERY);
+        submodule.checkRibbons(data, this);
+
+        int mode = getMode(data);
+        if (mode > 0) {
+            applyEffectTick(data);
+        }
+    }
+
+    @Override
+    public void applyEffectTick(ISpiritweb data)
+    {
+        performEffectServer(data);
+    }
+
+    private void performEffectServer(ISpiritweb data)
+    {
+        SpiritwebCapability playerSpiritweb = (SpiritwebCapability) data;
+        SandmasterySpiritwebSubmodule submodule = (SandmasterySpiritwebSubmodule) playerSpiritweb.spiritwebSubmodules.get(Manifestations.ManifestationTypes.SANDMASTERY);
+
+        LivingEntity living = data.getLiving();
+        Vec3 movement = living.getDeltaMovement();
+        if(movement.y > 0) return;
+        double distFromGround = MiscHelper.distanceFromGround(living);
```

Comment:
Was this also meant to use sand? Either from the jar or whitesand that's actually below the player?
Can't tell if I missing it somewhere

Reply:
I plan to make them all use sand somehow, but haven't implemented that yet

Reply:
It now requires sand in the form of a charged sand pouch in the belt curios slot